### PR TITLE
hyprnotify: init at version 0.6.2

### PR DESCRIPTION
**Pull request by FedX-sudo**
hyprnotify: init at version 0.6.2

## Description of changes

Add the hyprnotify notification utility. 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5592,6 +5592,8 @@ with pkgs;
 
   hyprshade = python311Packages.callPackage ../applications/window-managers/hyprwm/hyprshade { };
 
+  hyprnotify = callPackage ../by-name/hy/hyprnotify/package.nix { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:5595]:
```suggestion
```

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ];
+  buildInputs = [
+    alsa-lib
+  ];
+
+  vendorHash = "sha256-AZDtaiSNq7em876Q9f+YeDxboqVwA8IE9dDM6zggFXs=";
+
+  ldflags = [ "-s" "-w" ];
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/hy/hyprnotify/package.nix:29]:
```suggestion
  meta = {
```

tracking issue: #292468

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/top-level/all-packages.nix:5595]: Not sure what you mean. What I'm saying is to remove the entry in all-packages.nix that you added

Comment (FedX-sudo) [reply pkgs/top-level/all-packages.nix]: I was unaware that change had been made... Interesting...

File: pkgs/by-name/hy/hyprnotify/package.nix (hunk 2 above)
Comment (eclairevoyant) [reply pkgs/by-name/hy/hyprnotify/package.nix:29]: Don't this was fixed, btw

Comment (FedX-sudo) [reply pkgs/by-name/hy/hyprnotify/package.nix:29]: I am being yelled at when I remove that.

Comment (eclairevoyant) [reply pkgs/by-name/hy/hyprnotify/package.nix:29]: You have to put `lib.` where needed as well

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "codelif";
+    repo = "hyprnotify";
+    rev = "v${version}";
+    hash = "sha256-dAclsaY8gjHFyWe1/df98Tz5a1tJTZz2xvx8CZv1Q6Q=";
+  };
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/hy/hyprnotify/package.nix:21]:
```suggestion
  };

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = [
```

**Comment by SuperSandro2000** [comment]
already added by #322883

**Pull request closed without merging**